tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -40,9 +40,11 @@ private[scalanative] object LLVM {
       if (isLl || !Files.exists(objPath)) {
         val compiler = if (isCpp) config.clangPP.abs else config.clang.abs
         val stdflag =
-          if (isLl) Seq()
-          else if (isCpp) Seq("-std=c++11")
-          else Seq("-std=gnu11")
+          if (isLl) None
+          else if (isCpp) {
+            if (config.targetsWindows) Some("-std=c++14")
```

Comment:
C++14 or newer standard is needed to compile code using Windows API shipped with Windows 10 (used locally for development) or Server 2016 / 2019 used in the CI. 
I don't think we would be willing to support the older distributions, since their usage is now rather marginal.

As an example, header files included in our glue layer are using return type deduction (`auto`) which is not supported in std++11. There might have been other incompatibilities as well